modules/nuxeo-apidoc-webengine/src/main/resources/skin/resources/script/plotly.chart.js

```diff
@@ -0,0 +1,235 @@
+      positions.push(ismain ? 'auto': 'none');
+      if (ismain) {
+        mvalues.push(value.count);
+        mlabels.push(label);
+      }
+    });
+
+    var data = [{
+      values: mvalues,
+      labels: mlabels,
+      customdata: pieData,
+      type: 'pie',
+    }];
+    var layout =  getPieChartLayout(title);
+
+    var showsAll = mvalues.length == values.length;
+    if (!showsAll) {
+      data.push({
+        values: values,
+        labels: labels,
+        textposition: positions,
+        visible: false,
+        customdata: pieData,
+        type: 'pie',
+      });
+      Object.assign(layout, {
+        updatemenus: [{
+          type: 'buttons',
+          active: -1,
+          buttons: [{
+            args: [
+              { visible: [ false, true ] },
+            ],
+            args2: [
+              { visible: [ true, false ] },
+            ],
+            label: 'Show All',
+            method: 'update'
+          }]
+        }]
+      });
+    }
+
+    var div = document.getElementById(chartDivId);
+    div.textContent = '';
+    Plotly.newPlot(chartDivId, data, layout, getPieChartConfig());
+    div.on('plotly_click', data => showDetails(data, total, rootURL, chartDivId + 'Details'));
+  }
+
+  $.fn.plotlyStudioChart = (title, rootURL, stats, chartDivId) => {
+    var studio = stats.filter(stat => stat.fromStudio);
+    var notStudio = stats.filter(stat => !stat.fromStudio);
+    var pieData = [studio, notStudio];
+    var values = [countStats(studio), countStats(notStudio)];
+    var total = values.reduce((a, b) => a + b, 0);
+
+    var data = [{
+      values: values,
+      labels: ['From Studio', 'Not From Studio'],
+      customdata: pieData,
+      type: 'pie'
+    }];
+
+    var div = document.getElementById(chartDivId);
+    div.textContent = '';
+    Plotly.newPlot(chartDivId, data, getPieChartLayout(title), getPieChartConfig());
+    div.on('plotly_click', data => showDetails(data, total, rootURL, chartDivId + 'Details'));
+  }
+
+  function getPieChartLayout(title) {
+    return {
+      title: `${title}`,
+      grid: { rows: 1, columns: 2 },
+    };
+  }
+
+  function getPieChartConfig() {
+    return {
+      responsive: true,
+      displaylogo: false,
+      modeBarButtonsToRemove: ['hoverClosestPie', 'sendDataToCloud', 'toImage']
+    };
+  }
+
+  function filterStatsByType(stats, type) {
+    return stats.filter(stat => stat.codeType == type);
+  }
+
+  function countStats(stats) {
+    return stats.reduce((total, stat) => total + stat.numberOfContributions, 0);
+  }
+
+  function showDetails(clickData, total, rootURL, chartDetailsDivId) {
+    var point = clickData.points[0];
+    var tableStats = point.fullData.customdata[point.i];
+    var title = point.label;
+
+    var div = document.getElementById(chartDetailsDivId);
+    div.textContent = '';
+
+    var table = document.createElement('table');
+    table.classList.add("tablesorter");
```

Comment:
nit: again, with jQuery you have special helper chainable functions `.addClass('tablesorter').addClass('chart')` or even just `.addClass('tablesorter chart')`